api/src/routing.js

```diff
@@ -492,6 +494,12 @@ app.postJson('/api/v1/people', function(req, res) {
 app.get('/api/v1/person', person.v1.getAll);
 app.get('/api/v1/person/:uuid', person.v1.get);
 
+app.get('/api/v1/contact/id', contact.v1.getIds);
```

Comment:
That discussion happened in the parent ticket before we spun off the child isssue: https://github.com/medic/cht-core/issues/9544#issuecomment-2434471072

Reply:
Thanks @jkuester . Your argument here is that "we've already decided and your input is not welcome?"

Reply:
Sorry for being aggressive and confrontational in the above comment. 

I maintain my comment about `/api/v1/contact/id` being quite unsuggestive, we shouldn't need  thorough explanations and reasoning behind the naming choice in order for an api name to make sense.

Reply:
I was just trying to provide the context for the discussion that Sugat referenced. :grimacing:   

I am happy to continue the design discussion here to come to an agreed upon approach. It will just be most efficient if we all understand what was already said to get us here.  When starting work on new REST endpoints for the cht-datasource code, we chose to go with the pattern of _singular entity names_ (so `/api/v1/person` instead of `/api/v1/persons`).  When the endpoint can return `0-n` entities I do not really see a compelling reason to prefer either singular or plural (since either might make more sense depending on the context).  Two things seem clear to me though:

- Under normal circumstances, we should not duplicate endpoints for the same resource (e.g. having both  `/api/v1/contact/id` and `/api/v1/contact/ids`).  
- We should be consistent with our naming across our go-forward REST endpoints.  Either using singular or plural, but not mixing both.

Reply:
I agree with being consistent. 

I personally can't recall seeing APIs in the world that used this singular form, so for me this seems quite unintuitive.